### PR TITLE
Accept signet addresses for emergency

### PR DESCRIPTION
**Pull request by danielabrozzoni**
Signet addresses are parsed as testnet

### CODE REVIEW

File: src/common/config.rs [hunk 1]
```diff
@@ -243,7 +243,9 @@ impl Config {
 
             let emer_addr_net = stk_config.emergency_address.address().network;
             let bitcoind_net = config.bitcoind_config.network;
-            if emer_addr_net != bitcoind_net {
+            // All good, signet addresses have testnet type
+            if emer_addr_net == Network::Testnet && bitcoind_net == Network::Signet {
+            } else if emer_addr_net != bitcoind_net {
```
Comment (darosior) [inline src/common/config.rs:248]: This empty `if` clause really looks awkward to me. Also it's wrong as we still want to check for testnet.
I suggest:
```rust
// Testnet addresses are used for signet
if emer_addr_net != bitcoind_net && !(bitcoind_net == Network::Signet && emer_addr_net == Network::Testnet)
```

Comment (danielabrozzoni) [reply src/common/config.rs:248]: > Also it's wrong as we still want to check for testnet.

We are still checking for testnet, as you wouldn't enter the first if the bitcoind_net wasn't signet.
I agree the empty if is awkward, but it seems easier to understand to me. Anyways, I can change it, no problem

Comment (darosior) [reply src/common/config.rs:248]: Oh, right. Yeah my version is more verbose and less immediately clear..

Comment (darosior) [reply src/common/config.rs:248]: Maybe:
```rust
let signet_special_case = bitcoind_net == Network::Signet && emer_addr_net == Network::Testnet;
if emer_addr_net != bitcoind_net && !signet_special_case
```

Comment (danielabrozzoni) [reply src/common/config.rs:248]: Cool!

**Comment by darosior** [comment]
ACK b3171df522b396be5a6a6a680f853ad2fdaa73f1

**Pull request merged**